feature_engine/selection/probe_features_selector.py

```diff
@@ -0,0 +1,163 @@
+        y: array-like of shape (n_samples)
+           Target variable. Required to train the estimator.
+        """
+        # check input dataframe
+        X, y = check_X_y(X, y)
+
+        # find numerical and categorical variables
+        self.variables = _find_categorical_and_numerical_variables(X, None)
+
+        # if required excluded variables that not in the input dataframe
+        self._confirm_variables(X)
+
+        # check that there are more than 1 variable
+        self._check_variable_number()
+
+        # save input features
+        self._get_feature_names_in(X)
+
+        # generate 3 random variables
+        self.probe_features_ = self._generate_probe_features(X.shape[0])
+
+        # get random variable names
+        self.random_variables_ = self.probe_features_.columns
+
+        X_new = pd.concat([X, self.probe_features_], axis=1)
+
+        # train model with all variables including the random variables
+        model = cross_validate(
+            self.estimator,
+            X_new,
+            y,
+            cv=self.n_iter,
+            scoring=self.scoring,
+            return_estimator=True,
+        )
+
+        # Initialize a dataframe that will contain the list of the feature/coeff
+        # importance for each cross-validation fold
+        feature_importances_cv = pd.DataFrame()
+
+        # Populate the feature_importances_cv dataframe with columns containing
+        # the feature importance values for each model returned by the cross
+        # validation.
+        # There are as many columns as folds.
+        for i in range(len(model["estimator"])):
+            m = model["estimator"][i]
+            feature_importances_cv[i] = get_feature_importances(m)
+
+        # add the variables as the index to feature_importances_cv
+        feature_importances_cv.index = X_new.columns
+
+        # save feature importances as an attribute
+        self.feature_importances_ = feature_importances_cv
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Return a tuple comprised of the variables and the number of times
+        each variable was worse than all three random variables.
+        """
+        pass
+
+    def _generate_probe_features(self, n_obs: int) -> pd.DataFrame:
+        """
+        Returns a dataframe of 3 random variables.
+        """
+        # create 3 random variables
+        binary_var = np.random.randint(0, 2, n_obs)
+        uniform_var = np.random.uniform(0, 1, 1).tolist() * n_obs
+        gaussian_var = np.random.normal(0, 3, n_obs)
+
+        # create dataframe
+        df = pd.DataFrame()
+        df["rndm_binary_var"] = binary_var
+        df["rndm_uniform_var"] = uniform_var
+        df["rndm_gaussian_var"] = gaussian_var
+
+        return df
+
+
+    def _count_variables_importance_less_than_random_variables(self) -> Dict:
+        """
+        Count how many times a variable is less than the maximum value
```

Comment:
My understanding is that this transformer will select the variables whose importance greater than the importance of the random features.

Now, we have 3 random features, so we would have 2 importance values. What to do?

From this logic, I understand that you are taking the greatest of the importance values, correct?

Why count? I'd done if var_importance > max_var_random_vars then select otherwise drop.

Now, continuing thinking out loud, the random variable importance, will be random, so a conservative approach could be to take the average instead of the max.

Reply:
I included the count because I thought this would be an "exploratory class".

At this moment, I cannot recall precisely why, but I had the idea of comparing the variables' feature importances against multiple random variables, hence the `cv` parameter. I guess I wrote the code as such because of pure chance one random variable could be deemed more informative than variable X. We don't want to remove variable X because it was proven to be less informative than a variable on one unexpected occasion.

Therefore, the transformer performs `cv` number of iterations.

If we make this class a "standard" feature-engine transformer, which is to be used in a pipeline, we may want to count and add a `threshold` param.

The transformer will drop features that have frequencies of being less informative than the best-performing random variable above the passed `threshold` value.

Reply:
Let's update as per @glevv 's comment :)